### PR TITLE
Add broken dead links check v2

### LINKED ISSUE #72: Add broken/dead links check for markdown
## Is your feature request related to a OpenMS API Reference Documentation? Please describe.

No, maintenance automation.

- [x] I have attached screenshot describing the problem, if applicable.

## Describe the documentation enhancement you'd like
Add automated link/URL check for broken or dead links.

## Describe alternatives you've considered

This looks like a good GitHub action: https://github.com/marketplace/actions/markdown-link-check.

## Additional context

Would you be interested in adding this GitHub action? @greengypsy

### PR DESCRIPTION
**Pull request by ghost**
Fixes https://github.com/OpenMS/OpenMS-docs/issues/72.

## Describe the change

Adding broken/dead links check to staging branch.

## PR checklist

- [ ] I have added description of the change I'm proposing in the OpenMS Documentation.
- [ ] I have read and followed [OpenMS Documentation Contributing guidelines](CONTRIBUTING.md).
- [ ] I have attached a screenshot of the relevant area after this change.
- [ ] `CHANGELOG.md` is updated.
- [ ] I have added my name in [CONTRIBUTING.md](CONTRIBUTING.md#openms-documentation-contributors).

### CODE REVIEW

**Comment by tapaswenipathak** [comment]
is this ready?

**Comment by ghost** [comment]
> is this ready?

It should be.

File: .github/workflows/action.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Check Markdown links
+
+on:
+  pull_request:
+    schedule:
+      # Run everyday at 9:00 AM (See https://pubs.opengroup.org/onlinepubs/9699919799/utilities/crontab.html#tag_20_25_07)
+      - cron: "0 9 * * *"
```
Comment (tapaswenipathak) [inline .github/workflows/action.yml]: let's make it earlier or midnight 12:00 A.M.

File: .github/workflows/action.yml [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+jobs:
+  markdown-link-check:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@master
```
Comment (tapaswenipathak) [inline .github/workflows/action.yml:22]: and `develop`.

File: .github/workflows/action.yml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    - uses: gaurav-nelson/github-action-markdown-link-check@v1
+      with:
+        max-depth: -1
+        check-modified-files-only: 'yes'
+        config-file: 'config.json'
```
Comment (tapaswenipathak) [inline .github/workflows/action.yml:27]: `file-extension:` .rst and .md?

File: .github/workflows/config.json [hunk 4]
```diff
@@ -0,0 +1,7 @@
+{
+  "ignorePatterns": [
+    {
+      "pattern": "^http://docs.github"
```
Comment (tapaswenipathak) [inline .github/workflows/config.json:4]: + `pkgs.org`

File: .github/workflows/action.yml (hunk 1 above)
Comment (tapaswenipathak) [reply .github/workflows/action.yml]: to note: i think server(s) is in Tübingen, Germany timezone.

File: .github/workflows/action.yml [hunk 5]
```diff
@@ -0,0 +1,18 @@
+name: Check Markdown links
+
+on:
+  pull_request:
```
Comment (tapaswenipathak) [inline .github/workflows/action.yml:4]: i think, it should be:
```
on:
  pull_request:
  push:
    branches:
      - 'release-*'
      - 'develop'
      - 'staging'
      - 'main'
    tags:
      - 'v*'
 ```

File: .github/workflows/action.yml [hunk 6]
```diff
@@ -0,0 +1,18 @@
+name: Check Markdown links
```
Comment (tapaswenipathak) [inline .github/workflows/action.yml]: let's say, `Documentation URLs Check`

**Review by tapaswenipathak — COMMENTED** [review]
1. can you check if it is checking `.github/` dir, i think i didn't see it.
2. add a few screenshots?

otherwise, minor nits we discussed.

File: .github/workflows/action.yml (hunk 3 above)
Comment (ghost) [reply .github/workflows/action.yml:27]: done

File: .github/workflows/action.yml (hunk 5 above)
Comment (ghost) [reply .github/workflows/action.yml:4]: done

File: .github/workflows/action.yml (hunk 6 above)
Comment (ghost) [reply .github/workflows/action.yml]: done

File: .github/workflows/action.yml (hunk 2 above)
Comment (ghost) [reply .github/workflows/action.yml:22]: i don't know what you mean.

Comment (ghost) [reply .github/workflows/action.yml:22]: on:
  pull_request:
    push:
      branches:
        - 'release-*'
        - 'develop'
        - 'staging'
        - 'main'

**Pull request closed without merging**